### ISSUE TITLE
Complain about redundant ternaries?

### ISSUE THREAD

**Issue by janosh** [astral-sh/ruff#12516]
while refactoring i got this code and was fully expecting `ruff` to complain about the redundant ternary but it doesn't. maybe worth adding a rule for this?

```py
def test_the_answer():
    expected = 42 if below_311 else 42
    assert the_answer == expected
```

**Comment by charliermarsh** [comment]
That seems reasonable to me.

**Comment by njhearp** [comment]
I'm interested in working on this, but I'll be away for a week. If this issue is still open when I return, I would be happy to work on it.

**Comment by iamlucasvieira** [comment]
Hey @njhearp, have you started working on this? If not, I’d love to take it. Let me know!

**Comment by njhearp** [comment]
Hey @iamlucasvieira, you can take the issue. I tried working on the issue, but had difficulties because it is my first time trying to contribute.

**Comment by MichaReiser** [comment]
Implemented in https://github.com/astral-sh/ruff/pull/13218
